### ISSUE TITLE
Access length and value of models from .60

### ISSUE THREAD

**Issue by ogoffart** [sixtyfpsui/sixtyfps#98]
Suggested syntax:

```60
  property<[string]> model: ["set", "by", "the", "user"];
  property<string> current_value:  model.at(current_index);
  property<int> xxx:  model.length();
```

Alternative syntax

```60￼￼	
  property<string> current_value:  model[current_index];
  property<int> xxx:  model.length;
```

The problem is that we want to get binding updates when the model changes, which makes the implementation a bit tricky.

**Comment by tronical** [comment]
PR #580 covers the API part of this ticket - thanks @jamesblacklock !

**Comment by tronical** [comment]
I think I may have found a way to get hold of the `ModelNotify` without a temporary `ViewAbstraction` and a lot less plumbing: `ModelPeer` is public but can't be constructed. Attaching a lifetime to it and using a variant inside allows passing a callback reference. It's still abusing the term "Peer", but might be okay until we break compatibility.

**Comment by ogoffart** [comment]
(implemented in 0.1.6)
